### PR TITLE
[system test] - removal of class-wide parallelism support 

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Refactoring
- Documentation

### Description

This PR removes the order of the test suites. Also, this is the initial PR for removing:
1. support for class-wide parallelism which does not seem to fit our requirements (i.e., OrderTestSuites class, because it is not bounded to any other part)
2. BeforeAllOnce class, and re-place all test classes with branch new deployment of Cluster Operator.
3. Partially remove of ExecutionListener class, SuiteThreadController class and the logic inside AbstractST class
4. Remove completely annotations @ParallelSuite and @IsolatedSuite + StUtils clean-up

The reasons why we are gonna remove class-wide parallelism are:
1. It is hard to maintain (i.e., when one decides to create a test case it simply tests it in sequential mode. But to completely verify that everything works (parallelization) it has to be executed method-wide, where all test cases will be executed in parallel (this is usually done by us system-test owners but I think very few people are doing it apart from us. And lastly to fully prove that everything works one has to run it class-wide, which takes a lot of time where you have to check if there are no conflicts between classes and this is just a small piece of a huge maintaining nightmare.)
5. It is very costly for resources (i.e., running multiple classes takes a lot of resources)
6. It is very hard to troubleshoot a problem for other people than its creator (i.e., thread interleaving, problems that do not occur in sequential mode or method-wide because of conflicts between classes causing race conditions)

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by Frawless — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java [hunk 1]
```diff
@@ -278,14 +273,14 @@ public SetupClusterOperator runManualOlmInstallation(final String fromOlmChannel
     private void helmInstallation() {
         LOGGER.info("Install ClusterOperator via Helm");
         helmResource = new HelmResource(namespaceInstallTo, namespaceToWatch);
-        createCONamespaceIfNeeded();
+        createClusterOperatorNamespaceIfPossible();
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:276]: WHat will happen when the creation won't be possible? 😃

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 2]
```diff
@@ -77,8 +74,7 @@ public abstract class AbstractST implements TestSeparator {
     protected final ResourceManager resourceManager = ResourceManager.getInstance();
     protected final TestSuiteNamespaceManager testSuiteNamespaceManager = TestSuiteNamespaceManager.getInstance();
     private final SuiteThreadController parallelSuiteController = SuiteThreadController.getInstance();
-    protected SetupClusterOperator clusterOperator;
-    protected OlmResource olmResource;
+    protected SetupClusterOperator clusterOperator = SetupClusterOperator.getInstanceHolder();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:77]: Wouldn't be better to rename `getInstanceHolder` to `getInstance`?

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 3]
```diff
@@ -616,26 +580,7 @@ private void beforeEachMustExecute(ExtensionContext extensionContext) {
...
-                    // wait for parallel suites are done
-                    parallelSuiteController.waitUntilZeroParallelSuites(extensionContext);
-                    // wait for isolated suites
-                    parallelSuiteController.waitUntilEntryIsOpen(extensionContext);
-                } else if (StUtils.isParallelSuite(extensionContext) && Environment.isNamespaceRbacScope()) {
-                    cluster.setNamespace(Constants.INFRA_NAMESPACE);
-                }
-                clusterOperator = SetupClusterOperator.getInstanceHolder();
-            }
+            // nop
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:583]: reverse the condition?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java [hunk 4]
```diff
@@ -68,7 +64,7 @@ void testCorsOriginAllowed() {
 
         String url = bridgeUrl + "/consumers/" + groupId + "/instances/" + kafkaBridgeUser + "/subscription";
         String headers = BridgeUtils.addHeadersToString(additionalHeaders, Constants.KAFKA_BRIDGE_JSON_JSON);
-        String response = cmdKubeClient().namespace(namespace).execInPod(scraperPodName, "/bin/bash", "-c", BridgeUtils.buildCurlCommand(HttpMethod.OPTIONS, url, headers, "")).out().trim();
+        String response = cmdKubeClient().namespace(clusterOperator.getDeploymentNamespace()).execInPod(scraperPodName, "/bin/bash", "-c", BridgeUtils.buildCurlCommand(HttpMethod.OPTIONS, url, headers, "")).out().trim();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java:67]: I don't like the usage of `clusterOperator.getDeploymentNamespace()` in actions that are not related to CO. Isn't there any better way?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java [hunk 5]
```diff
@@ -52,18 +52,14 @@
...
-@ParallelSuite
 public class HttpBridgeKafkaExternalListenersST extends AbstractST {
     private static final String BRIDGE_EXTERNAL_SERVICE =  "shared-http-bridge-external-service";
-
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(HttpBridgeKafkaExternalListenersST.class.getSimpleName()).stream().findFirst().get();
     private final String producerName = "producer-" + new Random().nextInt(Integer.MAX_VALUE);
     private final String consumerName = "consumer-" + new Random().nextInt(Integer.MAX_VALUE);
 
-    @ParallelTest
+    @ParallelNamespaceTest("Creating a node port service and thus avoiding 409 error")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java:60]: What kind of 409 error?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 6]
```diff
@@ -43,12 +42,11 @@
 @Tag(BRIDGE)
 @Tag(REGRESSION)
 @KRaftNotSupported("UserOperator and scram-sha are not supported by KRaft mode and is used in this test class")
-@ParallelSuite
 class HttpBridgeScramShaST extends AbstractST {
     private static final Logger LOGGER = LogManager.getLogger(HttpBridgeScramShaST.class);
     private final String httpBridgeScramShaClusterName = "http-bridge-scram-sha-cluster-name";
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(HttpBridgeScramShaST.class.getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:49]: remove?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 7]
```diff
@@ -67,24 +65,24 @@ void testSendSimpleMessageTlsScramSha(ExtensionContext extensionContext) {
         // Create topic
         resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(httpBridgeScramShaClusterName, topicName)
             .editMetadata()
-                .withNamespace(namespace)
+                .withNamespace(clusterOperator.getDeploymentNamespace())
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:68]: Same as above.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java [hunk 8]
```diff
@@ -43,15 +42,14 @@
...
-@ParallelSuite
 class HttpBridgeTlsST extends AbstractST {
     private static final Logger LOGGER = LogManager.getLogger(HttpBridgeTlsST.class);
     private BridgeClients kafkaBridgeClientJob;
     private String kafkaClientsPodName;
     private String sharedKafkaUserName;
 
     private final String httpBridgeTlsClusterName = "http-bridge-tls-cluster-name";
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(HttpBridgeTlsST.class.getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:52]: Remove

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java [hunk 9]
```diff
@@ -68,24 +66,24 @@ void testSendSimpleMessageTls(ExtensionContext extensionContext) {
 
         resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(httpBridgeTlsClusterName, topicName)
             .editMetadata()
-                .withNamespace(namespace)
+                .withNamespace(clusterOperator.getDeploymentNamespace())
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:69]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlConfigurationST.java [hunk 10]
```diff
@@ -57,16 +57,13 @@
...
 public class CruiseControlConfigurationST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(CruiseControlConfigurationST.class);
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(CruiseControlConfigurationST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelNamespaceTest
     void testDeployAndUnDeployCruiseControl(ExtensionContext extensionContext) throws IOException {
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(namespace, extensionContext);
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(clusterOperator.getDeploymentNamespace(), extensionContext);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlConfigurationST.java:66]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java [hunk 11]
```diff
@@ -70,24 +70,21 @@
...
-
     @IsolatedTest
     @KRaftNotSupported("TopicOperator is not supported by KRaft mode and is used in this test class")
     void testAutoCreationOfCruiseControlTopicsWithResources(ExtensionContext extensionContext) {
         final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
 
         resourceManager.createResource(extensionContext, KafkaTemplates.kafkaWithCruiseControl(clusterName, 3, 3)
             .editMetadata()
-                .withNamespace(namespace)
+                .withNamespace(clusterOperator.getDeploymentNamespace())
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java:87]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/ConfigProviderST.java [hunk 12]
```diff
@@ -39,18 +39,15 @@
...
     private static final Logger LOGGER = LogManager.getLogger(ConfigProviderST.class);
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(ConfigProviderST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelNamespaceTest
     void testConnectWithConnectorUsingConfigAndEnvProvider(ExtensionContext extensionContext) {
         final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
         final String topicName = mapWithTestTopics.get(extensionContext.getDisplayName());
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(namespace, extensionContext);
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(clusterOperator.getDeploymentNamespace(), extensionContext);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/ConfigProviderST.java:50]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 13]
```diff
@@ -108,7 +105,7 @@ class KafkaST extends AbstractST {
     @ParallelNamespaceTest
     @KRaftNotSupported("EntityOperator is not supported by KRaft mode and is used in this test class")
     void testJvmAndResources(ExtensionContext extensionContext) {
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(namespace, extensionContext);
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(clusterOperator.getDeploymentNamespace(), extensionContext);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:108]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java [hunk 14]
```diff
@@ -46,13 +45,12 @@
...
 
 @Tag(REGRESSION)
-@ParallelSuite
 public class MultipleListenersST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(MultipleListenersST.class);
     private Object lock = new Object();
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(MultipleListenersST.class.getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java:53]: Remove

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java [hunk 15]
```diff
@@ -165,7 +163,7 @@ private void runListenersTest(ExtensionContext extensionContext, List<GenericKaf
         // exercise phase
         resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3)
             .editMetadata()
-                .withNamespace(namespace)
+                .withNamespace(clusterOperator.getDeploymentNamespace())
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java:166]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 16]
```diff
@@ -78,11 +77,10 @@
 
 @Tag(REGRESSION)
 @TestMethodOrder(OrderAnnotation.class)
-@ParallelSuite
 class LogSettingST extends AbstractST {
     private static final Logger LOGGER = LogManager.getLogger(LogSettingST.class);
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(LogSettingST.class.getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:83]: Remove

File: systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java [hunk 17]
```diff
@@ -199,34 +197,34 @@ void testKafkaLogSetting(ExtensionContext extensionContext) {
         LabelSelector kafkaSelector = KafkaResource.getLabelSelector(LOG_SETTING_CLUSTER_NAME, kafkaSsName);
         LabelSelector zkSelector = KafkaResource.getLabelSelector(LOG_SETTING_CLUSTER_NAME, zkSsName);
 
-        Map<String, String> eoPods = DeploymentUtils.depSnapshot(namespace, eoDepName);
-        Map<String, String> kafkaPods = PodUtils.podSnapshot(namespace, kafkaSelector);
-        Map<String, String> zkPods = PodUtils.podSnapshot(namespace, zkSelector);
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(clusterOperator.getDeploymentNamespace(), eoDepName);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java:200]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 18]
```diff
@@ -82,18 +82,15 @@
...
 
     private static final Pattern DEFAULT_LOG4J_PATTERN = Pattern.compile("^(?<date>[\\d-]+) (?<time>[\\d:,]+) (?<status>\\w+) (?<message>.+)");
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(LoggingChangeST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelNamespaceTest
     @SuppressWarnings({"checkstyle:MethodLength"})
     void testJSONFormatLogging(ExtensionContext extensionContext) {
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(namespace, extensionContext);
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(clusterOperator.getDeploymentNamespace(), extensionContext);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:93]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java [hunk 19]
```diff
@@ -41,24 +40,21 @@
...
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(ThrottlingQuotaST.class.getSimpleName()).stream().findFirst().get();
     private TestStorage sharedTestStorage;
 
     private KafkaAdminClientsBuilder adminClientsBuilder;
     private String scraperPodName = "";
 
     @ParallelTest
     void testThrottlingQuotasDuringAllTopicOperations(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext, namespace);
+        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java:57]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java [hunk 20]
```diff
@@ -78,15 +77,14 @@
...
-@ParallelSuite
 @KRaftNotSupported("TopicOperator is not supported by KRaft mode and is used in this test class")
 public class TopicST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(TopicST.class);
     private static final String KAFKA_CLUSTER_NAME = "topic-cluster-name";
     private static final String SCRAPER_NAME = KAFKA_CLUSTER_NAME + "-" + Constants.SCRAPER_NAME;
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(TopicST.class.getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:87]: Remove

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java [hunk 21]
```diff
@@ -97,30 +95,30 @@ void testMoreReplicasThanAvailableBrokers(ExtensionContext extensionContext) {
         int topicReplicationFactor = 5;
         int topicPartitions = 5;
 
-        KafkaTopic kafkaTopic = KafkaTopicTemplates.topic(KAFKA_CLUSTER_NAME, topicName, topicPartitions, topicReplicationFactor, 1, namespace).build();
+        KafkaTopic kafkaTopic = KafkaTopicTemplates.topic(KAFKA_CLUSTER_NAME, topicName, topicPartitions, topicReplicationFactor, 1, clusterOperator.getDeploymentNamespace()).build();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:98]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java [hunk 22]
```diff
@@ -708,18 +706,23 @@ void verifyTopicViaKafka(final String namespaceName, String topicName, int topic
...
 
     @BeforeAll
     void setup(ExtensionContext extensionContext) {
-        LOGGER.info("Deploying shared Kafka across all test cases in {} namespace", namespace);
+        this.clusterOperator = this.clusterOperator
+            .defaultInstallation(extensionContext)
+            .createInstallation()
+            .runInstallation();
+
+        LOGGER.info("Deploying shared Kafka across all test cases in {} namespace", clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java:721]:
```suggestion
        LOGGER.info("Deploying shared Kafka {}/{} across all test cases", clusterOperator.getDeploymentNamespace(), KAFKA_CLUSTER_NAME);
```

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 23]
```diff
@@ -80,27 +77,27 @@ void testUserWithNameMoreThan64Chars(ExtensionContext extensionContext) {
         String saslUserWithLongName = "sasl-user" + "abcdefghijklmnopqrstuvxyzabcdefghijklmnopqrstuvxyzabcdef"; // 65 character username
 
         // Create user with correct name
-        resourceManager.createResource(extensionContext, KafkaUserTemplates.tlsUser(namespace, userClusterName, userWithCorrectName).build());
+        resourceManager.createResource(extensionContext, KafkaUserTemplates.tlsUser(clusterOperator.getDeploymentNamespace(), userClusterName, userWithCorrectName).build());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:80]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java [hunk 24]
```diff
@@ -63,16 +63,13 @@
...
 class AlternativeReconcileTriggersST extends AbstractST {
     private static final Logger LOGGER = LogManager.getLogger(AlternativeReconcileTriggersST.class);
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(AlternativeReconcileTriggersST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelNamespaceTest
     @SuppressWarnings("checkstyle:MethodLength")
     void testManualTriggeringRollingUpdate(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext, namespace);
+        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java:72]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java [hunk 25]
```diff
@@ -73,19 +73,16 @@
...
     private static final Pattern ZK_SERVER_STATE = Pattern.compile("zk_server_state\\s+(leader|follower)");
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(RollingUpdateST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelNamespaceTest
     @Tag(ROLLING_UPDATE)
     @KRaftNotSupported("Zookeeper are not supported by KRaft mode and is used in this test case")
     void testRecoveryDuringZookeeperRollingUpdate(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext, namespace);
+        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/RollingUpdateST.java:85]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/security/OpaIntegrationST.java [hunk 26]
```diff
@@ -37,19 +36,16 @@
...
     private static final String OPA_GOOD_USER = "good-user";
     private static final String OPA_BAD_USER = "bad-user";
     private static final String CLUSTER_NAME = "opa-cluster";
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(OpaIntegrationST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelTest
     void testOpaAuthorization(ExtensionContext extensionContext) {
-        final TestStorage testStorage = new TestStorage(extensionContext, namespace);
+        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/security/OpaIntegrationST.java:48]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomAuthorizerST.java [hunk 27]
```diff
@@ -29,32 +28,27 @@
...
     private static final Logger LOGGER = LogManager.getLogger(CustomAuthorizerST.class);
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(CustomAuthorizerST.class.getSimpleName()).stream().findFirst().get();
-
     @ParallelTest
     @Tag(INTERNAL_CLIENTS_USED)
-    void testAclRuleReadAndWrite(ExtensionContext extensionContext) throws UnexpectedException {
-        final TestStorage testStorage = new TestStorage(extensionContext, namespace);
+    void testAclRuleReadAndWrite(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace());
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomAuthorizerST.java:44]: Same as above

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingAbstractST.java [hunk 28]
```diff
@@ -90,7 +90,7 @@ public abstract class TracingAbstractST extends AbstractST {
     private final String jaegerInstancePath = TestUtils.USER_PATH + "/../systemtest/src/test/resources/tracing/jaeger-instance.yaml";
     private final String jaegerOperatorPath = TestUtils.USER_PATH + "/../systemtest/src/test/resources/tracing/jaeger-operator.yaml";
 
-    private final String namespace = testSuiteNamespaceManager.getMapOfAdditionalNamespaces().get(this.getClass().getSimpleName()).stream().findFirst().get();
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingAbstractST.java:93]: Remove

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingAbstractST.java [hunk 29]
```diff
@@ -352,7 +352,7 @@ void doTestProducerConsumerStreamsConnectService(final ExtensionContext extensio
         configOfKafkaConnect.put("key.converter.schemas.enable", "false");
         configOfKafkaConnect.put("value.converter.schemas.enable", "false");
 
-        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(storageMap.get(extensionContext).getClusterName(), namespace, 1)
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(storageMap.get(extensionContext).getClusterName(), clusterOperator.getDeploymentNamespace(), 1)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingAbstractST.java:355]: Same as above

**Review by Frawless — COMMENTED** [review]
IIUC we still use cluster-wide operator and in that case we should use different namespaces for operands than infra-namespace that is used for operator. Thoughts?

**Comment by see-quick** [comment]
> IIUC we still use cluster-wide operator and in that case we should use different namespaces for operands than infra-namespace that is used for operator. Thoughts?

So for each test class, do we want to have a separate namespace? Meaning for example for SecurityST we will have:
1. infra namespace - where only CO will be deployed and then 
2. auxiliary namespace - for components (Kafka etc.) 

and plus do we want to implement this PR or not a different one?

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java (hunk 1 above)
Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:276]: It will be impossible only when we use two CO's in one namespace (so the second CO will not re-crease the already deployed namespace and remove the first one). That's currently one use case for this logic AFAIK.

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java (hunk 2 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:77]: yea sure

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java (hunk 5 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java:60]: `that service already exists => 409`

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java (hunk 4 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java:67]: well in @ParallelTest or @IsolatedTest where components are deployed in the same namespace where the operator is I think the okayish way to do it is this way. But this could definitely change if we decide to use two namespace (for operator and components)

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 30]
```diff
@@ -416,54 +413,59 @@ void testUOListeningOnlyUsersInSameCluster(ExtensionContext extensionContext) {
...
     void setup(ExtensionContext extensionContext) {
+        this.clusterOperator = this.clusterOperator
+            .defaultInstallation(extensionContext)
+            .createInstallation()
+            .runInstallation();
+
         resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(userClusterName, 1, 1)
             .editMetadata()
-                .withNamespace(namespace)
+                .withNamespace(clusterOperator.getDeploymentNamespace())
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:463]: out of curiosity, do we still need to specify the namespace, when it is now always same?

**Review by im-konge — APPROVED** [review]
Just two things. Thanks :)

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 30 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:463]: I think not, but maybe there are edge cases, which could switch the namespace context. (e.g., MultipleNamespaceST...). Anyway, I can try it and if it works, we can delete such occurrences in the code... :)

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:276]: so do we need `IfPossible` in the method name?

**Review by Frawless — APPROVED** [review]

**Comment by see-quick** [comment]
@strimzi-ci run tests --profile=acceptance --install-type=olm --ocp-version=4.13

**Comment by see-quick** [comment]
@strimzi-ci run tests --profile=acceptance --install-type=olm --ocp-version=4.13 --env=OLM_OPERATOR_VERSION=6.6.6

**Comment by see-quick** [comment]
> @strimzi-ci run tests --profile=acceptance --install-type=olm --ocp-version=4.13 --env=OLM_OPERATOR_NAME=strimzi-kafka-operator,OLM_OPERATOR_VERSION=6.6.6,OLM_SOURCE_NAME=strimzi-source,OLM_OPERATOR_DEPLOYMENT_NAME=strimzi-kafka-operator

Tests run: 36, Failures: 0, Errors: 0, Skipped: 0, Flakes: 1. I will resolve conflicts and then we are good to go I think

**Comment by im-konge** [comment]
Sounds good to me :) thanks
